pkg/sigproxy/signal/signal.go

```diff
@@ -0,0 +1,25 @@
+// Package signal provides helper functions for dealing with signals across
+// various operating systems.
+//
+// Forked from https://github.com/moby/moby/tree/37defbfd9b968f38e8e15dfa5f06d9f878bd65ba/pkg/signal
+package signal
+
+import (
+	"os"
+	"os/signal"
+)
+
+// CatchAll catches all signals and relays them to the specified channel.
+func CatchAll(sigc chan os.Signal) {
+	var handledSigs []os.Signal
+	for _, s := range SignalMap {
+		handledSigs = append(handledSigs, s)
+	}
+	signal.Notify(sigc, handledSigs...)
```

Comment:
Is there a reason this isn't just doing `signal.Notify(sigc)`? (which should catch all signals without an explicit list)

(Also https://github.com/opencontainers/runc/pull/2257, which I'm guessing you've probably already seen :smile:)

Reply:
copypasta from moby 😓